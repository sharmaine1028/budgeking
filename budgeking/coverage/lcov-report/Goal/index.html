
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for Goal</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../index.html">All files</a> Goal</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">6.35% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>27/425</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">3.68% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>6/163</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">2.83% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>4/141</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">6.5% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>27/415</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input oninput="onInput()" type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line low'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="EditGoal.js"><a href="EditGoal.js.html">EditGoal.js</a></td>
	<td data-value="1" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 1%"></div><div class="cover-empty" style="width: 99%"></div></div>
	</td>
	<td data-value="1" class="pct low">1%</td>
	<td data-value="100" class="abs low">1/100</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="45" class="abs low">0/45</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="31" class="abs low">0/31</td>
	<td data-value="1.01" class="pct low">1.01%</td>
	<td data-value="99" class="abs low">1/99</td>
	</tr>

<tr>
	<td class="file low" data-value="GenerateGoal.js"><a href="GenerateGoal.js.html">GenerateGoal.js</a></td>
	<td data-value="29.03" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 29%"></div><div class="cover-empty" style="width: 71%"></div></div>
	</td>
	<td data-value="29.03" class="pct low">29.03%</td>
	<td data-value="62" class="abs low">18/62</td>
	<td data-value="18.75" class="pct low">18.75%</td>
	<td data-value="32" class="abs low">6/32</td>
	<td data-value="17.64" class="pct low">17.64%</td>
	<td data-value="17" class="abs low">3/17</td>
	<td data-value="29.5" class="pct low">29.5%</td>
	<td data-value="61" class="abs low">18/61</td>
	</tr>

<tr>
	<td class="file low" data-value="GenerateOldGoal.js"><a href="GenerateOldGoal.js.html">GenerateOldGoal.js</a></td>
	<td data-value="5.55" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 5%"></div><div class="cover-empty" style="width: 95%"></div></div>
	</td>
	<td data-value="5.55" class="pct low">5.55%</td>
	<td data-value="18" class="abs low">1/18</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="8" class="abs low">0/8</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="10" class="abs low">0/10</td>
	<td data-value="5.55" class="pct low">5.55%</td>
	<td data-value="18" class="abs low">1/18</td>
	</tr>

<tr>
	<td class="file low" data-value="GoalHistory.js"><a href="GoalHistory.js.html">GoalHistory.js</a></td>
	<td data-value="3.22" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 3%"></div><div class="cover-empty" style="width: 97%"></div></div>
	</td>
	<td data-value="3.22" class="pct low">3.22%</td>
	<td data-value="31" class="abs low">1/31</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="13" class="abs low">0/13</td>
	<td data-value="3.44" class="pct low">3.44%</td>
	<td data-value="29" class="abs low">1/29</td>
	</tr>

<tr>
	<td class="file low" data-value="GoalsPage.js"><a href="GoalsPage.js.html">GoalsPage.js</a></td>
	<td data-value="4.76" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 4%"></div><div class="cover-empty" style="width: 96%"></div></div>
	</td>
	<td data-value="4.76" class="pct low">4.76%</td>
	<td data-value="84" class="abs low">4/84</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="22" class="abs low">0/22</td>
	<td data-value="3.44" class="pct low">3.44%</td>
	<td data-value="29" class="abs low">1/29</td>
	<td data-value="5" class="pct low">5%</td>
	<td data-value="80" class="abs low">4/80</td>
	</tr>

<tr>
	<td class="file low" data-value="NewGoal.js"><a href="NewGoal.js.html">NewGoal.js</a></td>
	<td data-value="0.94" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0.94" class="pct low">0.94%</td>
	<td data-value="106" class="abs low">1/106</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="50" class="abs low">0/50</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="32" class="abs low">0/32</td>
	<td data-value="0.96" class="pct low">0.96%</td>
	<td data-value="104" class="abs low">1/104</td>
	</tr>

<tr>
	<td class="file low" data-value="SaveToGoal.js"><a href="SaveToGoal.js.html">SaveToGoal.js</a></td>
	<td data-value="4.16" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 4%"></div><div class="cover-empty" style="width: 96%"></div></div>
	</td>
	<td data-value="4.16" class="pct low">4.16%</td>
	<td data-value="24" class="abs low">1/24</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="9" class="abs low">0/9</td>
	<td data-value="4.16" class="pct low">4.16%</td>
	<td data-value="24" class="abs low">1/24</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at 2022-07-22T07:34:21.618Z
            </div>
        <script src="../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../sorter.js"></script>
        <script src="../block-navigation.js"></script>
    </body>
</html>
    