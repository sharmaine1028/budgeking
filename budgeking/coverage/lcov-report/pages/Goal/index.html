
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for pages/Goal</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../index.html">All files</a> pages/Goal</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">6% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>26/433</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">3.63% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>6/165</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">3.49% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>5/143</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">6.14% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>26/423</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input oninput="onInput()" type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line low'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="EditGoal.js"><a href="EditGoal.js.html">EditGoal.js</a></td>
	<td data-value="0" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="100" class="abs low">0/100</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="45" class="abs low">0/45</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="31" class="abs low">0/31</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="99" class="abs low">0/99</td>
	</tr>

<tr>
	<td class="file low" data-value="GenerateGoal.js"><a href="GenerateGoal.js.html">GenerateGoal.js</a></td>
	<td data-value="34.37" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 34%"></div><div class="cover-empty" style="width: 66%"></div></div>
	</td>
	<td data-value="34.37" class="pct low">34.37%</td>
	<td data-value="64" class="abs low">22/64</td>
	<td data-value="18.75" class="pct low">18.75%</td>
	<td data-value="32" class="abs low">6/32</td>
	<td data-value="22.22" class="pct low">22.22%</td>
	<td data-value="18" class="abs low">4/18</td>
	<td data-value="34.92" class="pct low">34.92%</td>
	<td data-value="63" class="abs low">22/63</td>
	</tr>

<tr>
	<td class="file low" data-value="GenerateOldGoal.js"><a href="GenerateOldGoal.js.html">GenerateOldGoal.js</a></td>
	<td data-value="0" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="24" class="abs low">0/24</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="10" class="abs low">0/10</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="11" class="abs low">0/11</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="24" class="abs low">0/24</td>
	</tr>

<tr>
	<td class="file low" data-value="GoalHistory.js"><a href="GoalHistory.js.html">GoalHistory.js</a></td>
	<td data-value="0" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="31" class="abs low">0/31</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="13" class="abs low">0/13</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="29" class="abs low">0/29</td>
	</tr>

<tr>
	<td class="file low" data-value="GoalsPage.js"><a href="GoalsPage.js.html">GoalsPage.js</a></td>
	<td data-value="4.76" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 4%"></div><div class="cover-empty" style="width: 96%"></div></div>
	</td>
	<td data-value="4.76" class="pct low">4.76%</td>
	<td data-value="84" class="abs low">4/84</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="22" class="abs low">0/22</td>
	<td data-value="3.44" class="pct low">3.44%</td>
	<td data-value="29" class="abs low">1/29</td>
	<td data-value="5" class="pct low">5%</td>
	<td data-value="80" class="abs low">4/80</td>
	</tr>

<tr>
	<td class="file low" data-value="NewGoal.js"><a href="NewGoal.js.html">NewGoal.js</a></td>
	<td data-value="0" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="106" class="abs low">0/106</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="50" class="abs low">0/50</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="32" class="abs low">0/32</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="104" class="abs low">0/104</td>
	</tr>

<tr>
	<td class="file low" data-value="SaveToGoal.js"><a href="SaveToGoal.js.html">SaveToGoal.js</a></td>
	<td data-value="0" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="24" class="abs low">0/24</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="9" class="abs low">0/9</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="24" class="abs low">0/24</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at 2022-07-20T08:41:15.208Z
            </div>
        <script src="../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../sorter.js"></script>
        <script src="../../block-navigation.js"></script>
    </body>
</html>
    