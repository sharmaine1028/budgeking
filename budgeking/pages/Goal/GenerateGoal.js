import React, { useState, useEffect } from "react";
import {
  View,
  StyleSheet,
  Text,
  TouchableOpacity,
  Animated,
  Alert,
} from "react-native";
import { GreyLine } from "../../config/reusablePart";
import { Menu, MenuItem } from "react-native-material-menu";
import { MaterialCommunityIcons } from "@expo/vector-icons";
import { Header } from "../../config/reusableText";
import colours from "../../config/colours";
import { useNavigation } from "@react-navigation/native";
import { auth, db } from "../../config/firebase";
import { Ionicons } from "@expo/vector-icons";

function GenerateGoal({ doc, time, deleteItem, saveItem, editItem }) {
  const navigation = useNavigation();
  if (doc.deadline.seconds) {
    doc.deadline = new Date(doc.deadline.seconds * 1000);
  }
  const [isMenu, setIsMenu] = useState(false);
  const [showMore, setShowMore] = useState(false);
  const sharingEmails = doc.sharingEmails.filter(
    (item) => item !== auth.currentUser.email
  );

  useEffect(() => isOffTrack(), [doc.isOffTrack]);

  const getPercent = () => {
    const percent = (doc.currSavingsAmt / doc.target) * 100;
    if (percent > 100) {
      return "100%";
    } else {
      return (doc.currSavingsAmt / doc.target) * 100 + "%";
    }
  };

  const dateFormat = () => {
    const date = doc.deadline;
    const [day, month, year] = [
      date.getDate(),
      date.getMonth(),
      date.getFullYear(),
    ];

    const months = [
      "Jan",
      "Feb",
      "Mar",
      "Apr",
      "May",
      "Jun",
      "Jul",
      "Aug",
      "Sep",
      "Oct",
      "Nov",
      "Dec",
    ];

    return day.toString() + " " + months[month] + " " + year.toString();
  };

  const showMenu = () => {
    setIsMenu(true);
  };

  const hideMenu = (id) => {
    setIsMenu(false);
  };

  const deleteGoal = () => {
    let text;
    if (doc.createdBy === auth.currentUser.uid) {
      if (doc.isSharing) {
        text = "This goal will be deleted for all members and is irreversible.";
      } else {
        text = "This is an irreversible operation.";
      }
    } else {
      text = "This goal will only be deleted for you and not sharing members";
    }

    return Alert.alert("Are you sure?", text, [
      // "Yes button"
      {
        text: "Yes",
        onPress: () => {
          deleteItem(doc.id, time, doc);
        },
      },
      // The "No" button
      // Does nothing but dismiss the dialog when tapped
      {
        text: "No",
      },
    ]);
  };

  const isOffTrack = () => {
    const today = new Date();
    const deadline = doc.deadline;
    const dateCreated = new Date(doc.dateCreated.seconds * 1000);

    if (deadline < today) {
      db.collection("active goals").doc(doc.id).update({ isOffTrack: true });
      return;
    }
    // Get supposed amount based on frequency
    const years = today.getFullYear() - dateCreated.getFullYear();
    let supposedAmt;
    if (doc.frequency === "Yearly") {
      supposedAmt = doc.freqAmount * years;
    } else if (doc.frequency === "Monthly") {
      const months =
        years * 12 + today.getMonth() - dateCreated.getMonth() <= 0
          ? 0
          : years * 12 + today.getMonth() - dateCreated.getMonth();
      supposedAmt = doc.freqAmount * months;
    } else if (doc.frequency === "Weekly") {
      const msInWeek = 1000 * 60 * 60 * 24 * 7;
      const weeks = Math.round(Math.abs(today - dateCreated) / msInWeek);
      supposedAmt = doc.freqAmount * weeks;
    } else {
      const msInDay = 1000 * 3600 * 24;
      const days = Math.round(Math.abs(today - dateCreated) / msInDay);
      supposedAmt = doc.freqAmount * days;
    }

    // Compare supposed amount with curramount
    if (doc.currSavingsAmt < supposedAmt) {
      db.collection("active goals").doc(doc.id).update({ isOffTrack: true });
    } else {
      db.collection("active goals").doc(doc.id).update({ isOffTrack: false });
    }
  };

  return (
    <View key={doc.id} style={doc.isOffTrack ? styles.goalRed : styles.goal}>
      <View
        style={{
          flexDirection: "row",
          justifyContent: "space-between",
        }}
      >
        <Header
          text={`Save for ${doc.goalDescription}`}
          style={{ fontWeight: "bold" }}
        />

        <Menu
          visible={isMenu}
          anchor={
            <TouchableOpacity onPress={() => showMenu(doc.id)}>
              <MaterialCommunityIcons
                name="more"
                size={24}
                color={colours.black}
              />
            </TouchableOpacity>
          }
          onRequestClose={() => hideMenu(doc.id)}
        >
          <MenuItem
            onPress={() =>
              navigation.navigate("Edit Goal", {
                doc: doc,
                time: time,
                editItem: editItem,
              })
            }
          >
            Edit
          </MenuItem>

          <MenuItem
            onPress={() => {
              navigation.navigate("Save to Goal", {
                doc: doc,
                time: time,
                saveItem: saveItem,
              });
            }}
          >
            Save
          </MenuItem>
          <MenuItem onPress={() => deleteGoal(doc.id)}>Delete</MenuItem>
        </Menu>
      </View>

      {doc.isOffTrack ? (
        <Text style={styles.goalTagline}>
          You're behind schedule to save ${doc.freqAmount}{" "}
          {doc.frequency.toString().toLowerCase()}
        </Text>
      ) : (
        <Text style={styles.goalTagline}>
          You're on track to save ${doc.freqAmount}{" "}
          {doc.frequency.toString().toLowerCase()}
        </Text>
      )}

      <View style={styles.goalLine}>
        <MaterialCommunityIcons
          name="bullseye-arrow"
          size={24}
          color="black"
          style={{ flex: 0.1 }}
        />
        <View style={{ paddingLeft: 30, flex: 0.85 }}>
          <View style={[styles.progressBar]}>
            <Animated.View
              style={[
                StyleSheet.absoluteFill,
                {
                  backgroundColor: "#96D3FF",
                  width: getPercent(),
                  // width: (doc.currSavingsAmt / doc.target) * 100 + "%",
                  borderRadius: 5,
                },
              ]}
            ></Animated.View>
          </View>
          <View
            style={{
              flexDirection: "row",
              justifyContent: "space-between",
            }}
          >
            <Text style={styles.goalTagline}>
              ${doc.currSavingsAmt.toFixed(2)}
            </Text>
            <Text style={styles.goalTagline}>${doc.target.toFixed(2)}</Text>
          </View>
        </View>
      </View>
      <GreyLine />
      <View style={styles.goalLine}>
        <MaterialCommunityIcons
          name="clock-outline"
          size={24}
          color="black"
          style={{ flex: 0.1 }}
        />
        <Text style={{ flex: 0.3, paddingLeft: 30 }}> Deadline </Text>
        <Text style={{ flex: 0.55 }}>{dateFormat(doc.deadline)}</Text>

        {showMore ? (
          <TouchableOpacity onPress={() => setShowMore(false)}>
            <MaterialCommunityIcons
              name="chevron-up"
              size={24}
              color={colours.darkgrey}
            />
          </TouchableOpacity>
        ) : (
          <TouchableOpacity onPress={() => setShowMore(true)}>
            <MaterialCommunityIcons
              name="chevron-down"
              size={24}
              color={colours.darkgrey}
            />
          </TouchableOpacity>
        )}
      </View>

      {showMore && (
        <View>
          {doc.isSharing ? (
            <>
              <GreyLine />
              <View style={styles.goalLine}>
                <Ionicons
                  name="md-share-outline"
                  size={24}
                  color="black"
                  style={{ flex: 0.1 }}
                />
                <Text style={{ flex: 0.8, paddingLeft: 30 }}>
                  Goal shared with{" "}
                  <Text
                    style={{
                      fontWeight: "bold",
                      paddingLeft: 30,
                      flexWrap: "wrap",
                    }}
                  >
                    {sharingEmails
                      .map((email) => email.slice(0, email.indexOf("@")))
                      .join(", ")}
                  </Text>
                </Text>
              </View>
            </>
          ) : null}

          <>
            <GreyLine />
            <Text style={[styles.goalLine, { fontSize: 16 }]}>Notes </Text>
            <Text style={{ paddingLeft: 10 }}>
              {doc.notes === "" ? "None" : doc.notes}
            </Text>
          </>
        </View>
      )}
    </View>
  );
}

const styles = StyleSheet.create({
  goal: {
    backgroundColor: colours.lightBrown,
    borderRadius: 15,
    marginVertical: 5,
    padding: 10,
  },
  goalRed: {
    backgroundColor: colours.tomato,
    borderRadius: 15,
    marginVertical: 5,
    padding: 10,
  },
  goalLine: {
    flexDirection: "row",
    paddingLeft: 10,
    marginVertical: 5,
  },
  goalTagline: {
    color: colours.darkgrey,
    fontSize: 12,
    marginBottom: 10,
  },
  goalTitle: {
    color: "#6C757D",
    fontSize: 12,
  },
  progressBar: {
    flex: 1,
    height: 10,
    flexDirection: "row",
    backgroundColor: "#3F4243",
    borderColor: "#3F4243",
    borderRadius: 5,
  },
});

export default GenerateGoal;
