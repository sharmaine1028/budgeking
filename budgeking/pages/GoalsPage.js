import React from "react";
import { Text } from "react-native";

function GoalsPage(props) {
  return <Text>Howdy</Text>;
}

export default GoalsPage;
