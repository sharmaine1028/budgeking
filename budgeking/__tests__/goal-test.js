import { renderNoGoals } from "../pages/Goal/GoalsPage";
import renderer from "react-test-renderer";
import GenerateGoal from "../pages/Goal/GenerateGoal";
import { NavigationContainer } from "@react-navigation/native";
import { createStackNavigator } from "@react-navigation/stack";

test("goal renders correctly if empty database", async () => {
  const goals = [];
  const tree = renderer.create(
    goals.length !== 0
      ? goals.map((doc) => (
          <GenerateGoal
            key={doc.id}
            doc={doc}
            time={"short term"}
            deleteItem={this.deleteGoal}
            saveItem={this.saveToGoal}
            editItem={this.editGoal}
          />
        ))
      : renderNoGoals()
  );
  expect(tree).toMatchSnapshot();
});

test("goal renders correctly with data", async () => {
  const Stack = createStackNavigator();
  const MockNavigator = (
    <NavigationContainer>
      <Stack.Navigator>
        <Stack.Screen name="MockedScreen" component={GenerateGoal} />
      </Stack.Navigator>
    </NavigationContainer>
  );
  const goals = [
    (item = {
      id: 0,
      createdBy: "ywyztcPj3QcuDAahB9ZSGlBi2Ow2",
      createdByEmail: "a@gmail.com",
      currSavingsAmt: 5,
      dateCreated: new Date(2022, 7, 12, 20, 37, 13),
      deadline: new Date(2022, 7, 31, 23, 59, 59),
      freqAmount: "0.42",
      frequency: "Daily",
      goalDescription: "apples",
      isOffTrack: false,
      isSharing: true,
      notes: "",
      sharingEmails: ["b@gmail.com"],
      target: 10,
    }),
  ];

  jest.mock("@react-navigation/native", () => {
    const actualNav = jest.requireActual("@react-navigation/native");
    return {
      ...actualNav,
      useNavigation: () => ({
        navigate: jest.fn(),
        dispatch: jest.fn(),
      }),
    };
  });
  const tree = renderer
    .create(
      goals.length !== 0
        ? goals.map((doc) => (
            <GenerateGoal key={doc.id} doc={doc} time={"short term"} />
          ))
        : renderNoGoals()
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
