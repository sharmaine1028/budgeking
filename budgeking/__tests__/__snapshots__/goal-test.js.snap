// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`goal renders correctly with data 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#e3dad5",
      "borderRadius": 15,
      "marginVertical": 5,
      "padding": 10,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 15,
            "fontWeight": "400",
            "lineHeight": 15,
            "marginVertical": 5,
          },
          Object {
            "fontWeight": "bold",
          },
        ]
      }
    >
      Save for apples
    </Text>
    <View
      collapsable={false}
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text />
      </View>
      <Modal
        hardwareAccelerated={false}
        onRequestClose={[Function]}
        supportedOrientations={
          Array [
            "portrait",
            "portrait-upside-down",
            "landscape",
            "landscape-left",
            "landscape-right",
          ]
        }
        transparent={true}
        visible={false}
      />
    </View>
  </View>
  <Text
    style={
      Object {
        "color": "#6C757D",
        "fontSize": 12,
        "marginBottom": 10,
      }
    }
  >
    You're on track to save $
    0.42
     
    daily
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginVertical": 5,
        "paddingLeft": 10,
      }
    }
  >
    <Text />
    <View
      style={
        Object {
          "flex": 0.85,
          "paddingLeft": 30,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#3F4243",
              "borderColor": "#3F4243",
              "borderRadius": 5,
              "flex": 1,
              "flexDirection": "row",
              "height": 10,
            },
          ]
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "backgroundColor": "#96D3FF",
              "borderRadius": 5,
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": "50%",
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#6C757D",
              "fontSize": 12,
              "marginBottom": 10,
            }
          }
        >
          $
          5.00
        </Text>
        <Text
          style={
            Object {
              "color": "#6C757D",
              "fontSize": 12,
              "marginBottom": 10,
            }
          }
        >
          $
          10.00
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "borderBottomColor": "#E0E0E0",
        "borderBottomWidth": 2,
        "marginVertical": 3,
      }
    }
  />
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginVertical": 5,
        "paddingLeft": 10,
      }
    }
  >
    <Text />
    <Text
      style={
        Object {
          "flex": 0.3,
          "paddingLeft": 30,
        }
      }
    >
       Deadline 
    </Text>
    <Text
      style={
        Object {
          "flex": 0.55,
        }
      }
    >
      31 Aug 2022
    </Text>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text />
    </View>
  </View>
</View>
`;

exports[`goal renders no goals yet if empty database 1`] = `
<Text
  style={
    Array [
      undefined,
      Object {
        "alignSelf": "center",
        "marginTop": 20,
      },
    ]
  }
>
  No Goals yet
</Text>
`;
